pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transform for cancelling adjacent inverse gates in quantum circuits."""
+
+from pennylane import apply
+from pennylane.wires import Wires
+from pennylane.transforms import qfunc_transform
+
+from .optimization_utils import _find_next_gate
+
+
+@qfunc_transform
+def cancel_inverses(tape):
+    """Quantum function transform to remove any operations that are applied next to their
+    (self-)inverse.
+
+    Args:
+        tape (.QuantumTape): A quantum tape.
```

Comment:
I think I left a similar comment on another PR, but I would suggest changing this to `qfunc`, to match how users would use it
```suggestion
        qfunc (function): A quantum function.
```